src/TreeHouse/Queue/Message/Provider/AmqpMessageProvider.php

```diff
@@ -51,4 +51,20 @@ public function nack(Message $message, $requeue = false)
     {
         $this->queue->nack($message->getId(), $requeue ? AMQP_REQUEUE : null);
     }
+
+    /**
+     * @inheritdoc
+     */
+    public function consume(callable $callback)
+    {
+        $this->queue->consume(function(\AMQPEnvelope $envelope) use ($callback) {
+            $id    = $envelope->getDeliveryTag();
+            $body  = $envelope->getBody();
+            $props = new MessageProperties($envelope->getHeaders());
+
+            $message = new Message($body, $props, $id);
+
+            $callback($message);
```

Comment:
Hm, good call. I missed that and thought the consume method would only block until one message was received and then release it back to php. In our setup the callback should always return false.